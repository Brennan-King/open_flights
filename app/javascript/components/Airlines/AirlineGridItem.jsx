import React from 'react';
import PropTypes from 'prop-types';
// eslint-disable-next-line no-unused-vars
import {BrowserRouter as Router, Link} from 'react-router-dom';
import './AirlineGridItem.scss';

const AIRLINE_GRID_ITEM_BUTTON_TEXT = 'View Airline';

const propTypes = {
  /** The url of the airline image. */
  imageUrl: PropTypes.string,
  /** The name of the airline. */
  name: PropTypes.string,
  /** The average score rating for the airline. */
  avgScore: PropTypes.number,
  /** The name formatted with dashes between words. */
  slug: PropTypes.string,
};

const propDefaults = {
  imageUrl: '',
  name: '',
  avgScore: 0,
  slug: '',
};

/**
 * Constructs an airline grid item to be placed within a grid.
 * @param {object} props the props used to configure the airline grid item
 * @return {node} a div containing the configured AirglinGridItem
 */
const AirlineGridItem = (props) => {
  const {imageUrl, name, avgScore, slug} = props;

  const airlineAverageScoreText = `Average Rating: ${avgScore}`;

  return (
    <div className="airline-grid-item-card">
      <div className="airline-grid-item-logo-container">
        <img className='airline-grid-item-logo' src={imageUrl} alt={name} />
      </div>
      <div className="airline-grid-item-name">{name}</div>
      <div className="airline-score">{airlineAverageScoreText}</div>
      <div className="airline-grid-item-link-container">
        <Link className="airline-grid-item-link" to={`/airlines/${slug}`}>
          {AIRLINE_GRID_ITEM_BUTTON_TEXT}
        </Link>
      </div>
    </div>
  );
};

AirlineGridItem.propTypes = propTypes;
AirlineGridItem.defaultProps = propDefaults;
export default AirlineGridItem;
