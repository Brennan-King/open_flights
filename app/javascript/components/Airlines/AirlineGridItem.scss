.airline-grid-item-card {
  border: 2px solid #000;
  border-radius: 5px;
  background: #3796cb;
  text-align: center;
  box-shadow: 10px 10px 5px grey;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  color: #fff;
}

.airline-grid-item-logo-container {
  width: 50px;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  padding-top: 10px;
}

.airline-grid-item-logo {
  height: 50px;
  width: 50px;
  border-radius: 100%;
  border: 1px solid #fff
}

.airline-grid-item-name {
  padding: 20px 0 10px 0;
}

.airline-grid-item-link-container {
  margin: 30px 0 20px 0;
  height: 50px;
}

.airline-grid-item-link {
  color: #fff;
  background: #000;
  border-radius: 4px;
  padding: 10px 50px;
  border: 1px solid #000;
  width: 100%;
  text-decoration: none;
}