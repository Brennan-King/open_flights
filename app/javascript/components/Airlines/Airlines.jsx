import React from 'react';

const Airlines = () => {
  return <div>This is the airlines index teste view</div>;
};

export default Airlines;
