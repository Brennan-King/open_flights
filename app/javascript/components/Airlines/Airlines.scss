.airlines-container {
  text-align: center;
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
}

.airlines-header {
  padding: 25px 25px 25px 25px;
  font-size: 80px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.airlines-sub-header {
  font-weight: 300;
  font-size: 26px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.airlines-grid {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 20px;
  width: 100%;
  padding: 20px;
}

body {
  background-color: darkgray;
}