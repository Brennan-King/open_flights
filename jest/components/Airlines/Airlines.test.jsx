/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import Airlines from '../../../app/javascript/components/Airlines/Airlines';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {act} from 'react-dom/test-utils';
import * as axios from 'axios';

configure({adapter: new Adapter()});
jest.mock('axios');

/**
 * Test suite for Airlines component
 */
describe('Airlines', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render an Airlines component when the service successfully returns data', async () => {
    let component;
    axios.get.mockImplementation(() => Promise.resolve({data: [{
      id: '123',
      name: 'testAirlineName',
      image_url: 'testAirlineImageUrl',
      avg_score: 4.5,
      reviews: [{
        title: 'testReviewTitle',
        description: 'testReviewDescription',
        score: 3,
        id: '456',
      }],
    }]}));

    await act(async () => {
      component = mount(
          <Airlines />,
      );
    });
    await component.update();

    expect(component).toMatchSnapshot();
  });

  it('should render an Airlines component when the service unsuccessfully returns data', async () => {
    let component;
    axios.get.mockImplementation(() => Promise.reject(new Error('Service call failed')));

    await act(async () => {
      component = mount(
          <Airlines />,
      );
    });
    await component.update();

    expect(component).toMatchSnapshot();
  });
});
