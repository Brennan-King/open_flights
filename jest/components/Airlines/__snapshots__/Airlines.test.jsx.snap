// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airlines should render an Airlines component when the service successfully returns data 1`] = `null`;

exports[`Airlines should render an Airlines component when the service unsuccessfully returns data 1`] = `
<Airlines>
  <div
    className="airlines-container"
  >
    <div
      className="airlines-header"
    >
      Open Flights
    </div>
    <div
      className="airlines-sub-header"
    >
      Honest, unbiased airlines reviews
    </div>
    <div
      className="airlines-grid"
    />
  </div>
</Airlines>
`;
